### DISCUSSION TITLE
Feature request: Node power control

### DISCUSSION THREAD

**Discussion by Munsio** [dougiteixeira/proxmoxve#119] [Feature request] [answered]
Currently there is a power control for VMs and LXC but that functionality is missing for nodes themself.

As some of my servers in the cluster are not running all the times (energy prices in Europe) it would be great to have the controls for start/shutdown/reboot for nodes too.

In a cluster the nodes can be started with proxmox internal wake on lan via the API.

- Internal WOL: https://pve.proxmox.com/pve-docs/api-viewer/#/nodes/{node}/wakeonlan
- Getting the MAC for WOL: https://pve.proxmox.com/pve-docs/api-viewer/#/nodes/{node}/config
- Reboot/Shutdown: https://pve.proxmox.com/pve-docs/api-viewer/#/nodes/{node}/status - Switch to POST

**Comment by dougiteixeira** [comment]
At the moment I don't have the time to invest in these improvements, but I'm happy to merge them if you want to contribute.

**Comment by VeeeneX** [comment, marked as answer]
@Munsio I've added partial implementation in #38 but I didn't include the wakeup/start. Wake on lan can be done with HAS https://www.home-assistant.io/integrations/wake_on_lan/, and this is how it looks like:

![image](https://user-images.githubusercontent.com/5690163/218263062-8b9501bd-bb1f-453f-9fc9-f4919d42c698.png)

**Reply by lntrpdx** [reply]
Hi, I have added wake-on-lan button as I needed a way to start my proxmox node that seat on a different sub network that my home assistant container. see PR https://github.com/dougiteixeira/proxmoxve/pull/289

**Comment by dougiteixeira** [comment]
Released version 1.7.1 with changes.
